app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/tutorial/TutorialScreen.kt

```diff
@@ -11,236 +12,260 @@ import androidx.compose.foundation.layout.Spacer
-                }
-
-                item {
-                    val alpha = remember(state) {
-                        if (state.isLastPage) {
-                            0f
-                        } else {
-                            1f
-                        }
-                    }
-                    BitwardenTextButton(
-                        modifier = Modifier
-                            .fillMaxWidth()
-                            .alpha(alpha),
-                        isEnabled = !state.isLastPage,
-                        label = stringResource(id = R.string.skip),
-                        onClick = remember(viewModel) {
-                            {
-                                viewModel.trySendAction(TutorialAction.SkipClick)
-                            }
-                        },
-                    )
-                }
+@Composable
+private fun TutorialScreenContent(
+    state: TutorialState,
+    pagerState: PagerState,
+    onPagerSwipe: (Int) -> Unit,
+    onDotClick: (Int) -> Unit,
+    continueClick: (Int) -> Unit,
+    skipClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    LaunchedEffect(pagerState.currentPage) {
+        onPagerSwipe(pagerState.currentPage)
+    }
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier,
+    ) {
+        Spacer(modifier = Modifier.weight(1f))
+
+        HorizontalPager(state = pagerState) { index ->
+            if (LocalConfiguration.current.isPortrait) {
+                TutorialScreenPortrait(
+                    state = state.pages[index],
+                    modifier = Modifier.standardHorizontalMargin(),
+                )
+            } else {
+                TutorialScreenLandscape(
+                    state = state.pages[index],
+                    modifier = Modifier
+                        .standardHorizontalMargin(landscape = LANDSCAPE_HORIZONTAL_MARGIN),
+                )
             }
         }
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        IndicatorDots(
+            selectedIndexProvider = { state.index },
+            totalCount = state.pages.size,
+            onDotClick = onDotClick,
+            modifier = Modifier
+                .padding(bottom = 12.dp)
+                .height(44.dp),
+        )
+
+        BitwardenFilledTonalButton(
+            label = state.actionButtonText,
+            onClick = { continueClick(state.index) },
+            modifier = Modifier
+                .standardHorizontalMargin(landscape = LANDSCAPE_HORIZONTAL_MARGIN)
+                .fillMaxWidth(),
+        )
+
+        BitwardenTextButton(
+            isEnabled = !state.isLastPage,
+            label = stringResource(id = R.string.skip),
+            onClick = skipClick,
+            modifier = Modifier
+                .standardHorizontalMargin(landscape = LANDSCAPE_HORIZONTAL_MARGIN)
+                .fillMaxWidth()
+                .alpha(if (state.isLastPage) 0f else 1f)
```

Comment:
This is technically still clickable, should we disable it too?

Reply:
> isEnabled = !state.isLastPage,

Won't this prevent click events?

Reply:
Ohh, I missed that. We should be good